### PR TITLE
[MySQL] Implement "hash_id" for the MySQLDataSource

### PR DESCRIPTION
**Pull request by timgrein**
## Closes https://github.com/elastic/enterprise-search-team/issues/4258

This PR adds an implementation for "hash_id" for MySQL as future changes will immensely increase the likelihood of violating the length constraint of elasticsearch ids. 

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
~- [ ] Considered corresponding documentation changes~
~- [ ] Contributed any configuration settings changes to the configuration reference~

### CODE REVIEW

File: connectors/sources/mysql.py [hunk 1]
```diff
@@ -317,6 +318,11 @@ def _mysql_client(self):
     def advanced_rules_validators(self):
         return [MySQLAdvancedRulesValidator(self)]
 
+    @classmethod
```
Comment (tarekziade) [inline connectors/sources/mysql.py:321]: why not a simple function in utils? or a regular method?

**Comment by tarekziade** [comment]
This new function is not used, just added. Any reason not to use it within this patch?

File: connectors/sources/mysql.py (hunk 1 above)
Comment (wangch079) [reply connectors/sources/mysql.py:321]: I also think this is not necessary a connector-specific implementation: https://github.com/elastic/connectors-python/pull/721#discussion_r1158086080

Comment (timgrein) [reply connectors/sources/mysql.py]: @wangch079 @tarekziade I've created a now a `hash_id` function in `utils.py`, which will also be used as default implementation inside `BaseDataSource` so you don't have to worry about it as someone building a new connector.

**Comment by timgrein** [comment]
> This new function is not used, just added. Any reason not to use it within this patch?

It's used here: https://github.com/elastic/connectors-python/blob/main/connectors/sync_job_runner.py#L241

File: connectors/sources/mysql.py (hunk 1 above)
Comment (wangch079) [reply connectors/sources/mysql.py]: I meant, you don't have to implement a `hash_id` method in `BaseDataSource` class

**Review by artem-shelkovnikov — DISMISSED** [review]

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**